opb-crank/src/scripts/crank.ts

```diff
@@ -0,0 +1,227 @@
+
+const cluster = CLUSTER || 'mainnet';
+const interval = 10; // seconds between crank attempts
+const maxUniqueAccounts = parseInt(MAX_UNIQUE_ACCOUNTS || '10', 10);
+const consumeEventsLimit = new BN(CONSUME_EVENTS_LIMIT || '30');
+const priorityMarkets = PRIORITY_MARKETS ? PRIORITY_MARKETS.split(',') : [];
+const priorityQueueLimit = parseInt(PRIORITY_QUEUE_LIMIT || '100', 10);
+const CuLimit = parseInt(PRIORITY_CU_LIMIT || '50000', 10);
+const maxTxInstructions = parseInt(MAX_TX_INSTRUCTIONS || '1', 10);
+const serumProgramId = new PublicKey(
+  PROGRAM_ID || cluster === 'mainnet'
+    ? 'srmqPvymJeFKQ4zGQed1GFppgkRHL9kaELCbyksJtPX'
+    : 'EoTcMgcDRTJVZDMZWBoU6rhYHZfkNTVEAfz3uUJRcYGj',
+);
+const payer = Keypair.fromSecretKey(Uint8Array.from(readKeypair()));
+
+const log: Logger = new Logger({
+  name: 'openbook-cranker', displayFunctionName: false, displayFilePath: 'hidden', minLevel: 'info',
+});
+
+log.info(payer.publicKey.toString());
+
+const connection = new Connection(RPC_URL!, 'processed' as Commitment);
+
+// blockhash loop
+let recentBlockhash: BlockhashWithExpiryBlockHeight;
+try {
+  connection.getLatestBlockhash(
+    'finalized',
+  ).then((blockhash) => {
+    recentBlockhash = blockhash;
+  });
+} catch (e) {
+  log.error(`Couldn't get blockhash: ${e}`);
+}
+setInterval(async () => {
+  try {
+    recentBlockhash = await connection.getLatestBlockhash('finalized');
+  } catch (e) {
+    log.error(`Couldn't get blockhash: ${e}`);
+  }
+}, 1000);
+
+async function trytoCrank(spotMarkets: Market[]) {
+  try {
+    const eventQueuePks = spotMarkets.map(
+      (market) => market.decoded.eventQueue,
+    );
+
+    // pass a minimum Context Slot to GMA
+    let minContextSlot = 0;
+    const crankInstructionsQueue: TransactionInstruction[] = [];
+    const instructionBumpMap = new Map();
+
+    const eventQueueAccts = await getMultipleAccounts(
+      connection,
+      eventQueuePks,
+      'processed',
+      minContextSlot,
+    );
+
+    // increase the minContextSlot to avoid processing the same slot twice
+    minContextSlot = eventQueueAccts[0].context.slot + 1;
+
+    for (let i = 0; i < eventQueueAccts.length; i++) {
+      const { accountInfo } = eventQueueAccts[i];
+      const events = decodeEventQueue(accountInfo.data);
+
+      if (events.length === 0) {
+        continue;
+      }
+
+      const accounts: Set<string> = new Set();
+      for (const event of events) {
+        accounts.add(event.openOrders.toBase58());
+
+        // Limit unique accounts to first 10
+        if (accounts.size >= maxUniqueAccounts) {
+          break;
+        }
+      }
+
+      const openOrdersAccounts = [...accounts]
+        .map((s) => new PublicKey(s))
+        .sort((a, b) => a.toBuffer().swap64().compare(b.toBuffer().swap64()));
+
+      // coinFee & pcFee are redundant for cranking.
+      // Instead, we pass spotMarkets[i]['_decoded'].eventQueue
+      // using duplicate accounts will reduce transaction size
```

Comment:
We shouldnt be caring about transaction size since we arent doing too much cranking. Fine to leave though